src/core/ActionResult.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @fileoverview
+ * @author Sebastian Castro - sebastian.castro@picknik.ai
+ */
+
+var assign = require('object-assign');
+
+/**
+ * An ActionResult is returned from sending a ROS 2 action goal.
+ *
+ * @constructor
+ * @template TResult
+ */
+class ActionResult {
```

Comment:
Is there anywhere using an `instanceof` comparison on this, or can we remove these weird stub types in favor of just letting the user's generic type arguments pass all the way through? I don't really like this model that `Message` set up, but didn't want to API-break it in adopting TypeScript.

Same question about ActionGoal. I think these classes are unnecessary.

Reply:
The place they are used is in snippets like these: https://github.com/RobotWebTools/roslibjs/blob/ros2-actions/src/core/Action.js#L72-L74

Would we be able to just directly use the `TResult` generic here (and in equivalent places) without doing `new ActionResult(message.values)`?

Reply:
Yeah, we can just pass `message.values` through. This constructor is basically just doing a copy of `message.values`.